instant-xml/src/lib.rs

```diff
@@ -10,15 +10,165 @@ pub use macros::{FromXml, ToXml};
+    ) -> Result<(), Error> {
+        let value = match self {
+            true => "true",
+            false => "false",
+        };
+
+        let field_data = match field_data {
+            Some(field_data) => field_data,
+            None => &FieldData {
+                field_name: "bool",
+                field_attribute: None,
+            },
+        };
+
+        serializer.add_open_tag(field_data);
+        serializer.output.push_str(value);
+        serializer.add_close_tag(field_data);
+
+        Ok(())
+    }
+}
+
+impl ToXml for String {
+    fn serialize<'xml>(
+        &self,
+        serializer: &mut Serializer,
+        field_data: Option<&mut FieldData>,
+    ) -> Result<(), Error> {
+        let field_data = match field_data {
+            Some(field_data) => field_data,
+            None => &FieldData {
+                field_name: "String",
+                field_attribute: None,
+            },
+        };
+
+        serializer.add_open_tag(field_data);
+        serializer.output.push_str(self);
+        serializer.add_close_tag(field_data);
+        Ok(())
+    }
+}
+
+pub struct Serializer<'xml> {
+    pub parent_prefixes: &'xml mut BTreeSet<&'xml str>,
+    pub output: &'xml mut String,
+}
+
+impl<'xml> Serializer<'xml> {
+    fn add_open_tag(&mut self, field_data: &FieldData) {
+        match field_data.field_attribute {
+            Some(FieldAttribute::Prefix(prefix)) => {
+                self.output.push('<');
+                self.output.push_str(prefix);
+                self.output.push(':');
+                self.output.push_str(field_data.field_name);
+                self.output.push('>');
+            }
+            Some(FieldAttribute::Namespace(namespace)) => {
+                self.output.push('<');
+                self.output.push_str(field_data.field_name);
+                self.output.push_str(" xmlns=\"");
+                self.output.push_str(namespace);
+                self.output.push_str("\">");
+            }
+            _ => {
+                self.output.push('<');
+                self.output.push_str(field_data.field_name);
+                self.output.push('>');
+            }
+        }
+    }
+
+    fn add_close_tag(&mut self, field_data: &FieldData) {
+        match field_data.field_attribute {
+            Some(FieldAttribute::Prefix(prefix)) => {
+                self.output.push_str("</");
+                self.output.push_str(prefix);
+                self.output.push(':');
+                self.output.push_str(field_data.field_name);
+                self.output.push('>');
+            }
+            _ => {
+                self.output.push_str("</");
+                self.output.push_str(field_data.field_name);
+                self.output.push('>');
+            }
+        }
     }
 }
 
+pub enum FieldAttribute<'xml> {
+    Prefix(&'xml str),
+    Namespace(&'xml str),
+}
+
+pub struct FieldData<'xml> {
+    pub field_name: &'xml str,
```

Comment:
Let's rename `FieldData` to `FieldContext` and rename the `field_` prefixes from the field names (since these are kinda redundant anyway).